CONTRIBUTING.md

```diff
@@ -44,6 +46,42 @@ then it is okay to set them to 1 at the beginning of the file.
 #set-options "--fuel 1 --ifuel 1"
 ```
 
+## Argument ordering
+
+Function arguments are written in some order,
+for example in
+
+```fstar
+val map:
+  #a:Type ->
+  f:(a -> b) -> l:list a ->
+  list b
+```
+
+the first argument is `#a:Type`,
+the second one is `f:(a -> b)`
+and the third one is `l:list a`.
+
+Arguments should be ordered from the most generic one to the most specific one.
+There are several reasons for that:
+
+- this helps with curryfication,
+  which is useful when doing proofs in F\*
+  (because it is much easier to do proofs on `map f` than `fun l -> map f l`)
+- this helps having a consistent argument order throughout the project
+
+In the example of `map`,
+the most generic parameter is `#a:Type`,
+because it cannot be changed without changing `f:(a -> b)` or `l:list a`.
+The second most generic parameter is `f:(a -> b)`
+because often one function is applied to many different lists.
+Finally, `l:list a` is the least generic argument (or, the most specific one).
+Hence the arguments of the `map` function above are well-ordered.
+
+Often, several arguments may be as generic as other arguments,
+in that case some arbitrary order must be chosen between them,
+but that order should be consistent across functions.
```

Comment:
Do you follow a rule for ordering similar generic arguments in DY*? If so, you should probably mention it here.

Reply:
Indeed, I thought I had nothing else to say about that but I added one more paragraph!